dataprofiler/profilers/profile_builder.py

```diff
@@ -1170,11 +1173,23 @@ def _add_error_checks(self, other):
         StructuredProfiler specific checks to ensure two profiles can be added
         together.
         """
-        if set(self._profile) != set(other._profile):
+        schemas_differ = True
+        # Schemas must completely match if both contain duplicates
+        if self._duplicate_cols_present and other._duplicate_cols_present \
+                and self._col_name_to_idx == other._col_name_to_idx:
+            schemas_differ = False
+        # Can allow permutation of unique columns
+        elif not self._duplicate_cols_present and not \
+                other._duplicate_cols_present \
+                and self._col_name_to_idx.keys() == \
+                other._col_name_to_idx.keys():
+            schemas_differ = False
+
+        if schemas_differ:
             raise ValueError('Profiles do not have the same schema.')
-        elif not all([isinstance(other._profile[p_name],
-                                 type(self._profile[p_name]))
-                      for p_name in self._profile]):  # options check
+        elif not all([isinstance(other._profile[idx],
+                                 type(self._profile[idx]))
+                      for idx in range(len(self._profile))]):  # options check
```

Comment:
How redundant is this with the checks in the update chunk? e.g. can we reduce these both to use the same func ultimately?